### ISSUE TITLE
Toast component stale messages state issue, while quickly calling toastRef.show

### ISSUE THREAD

**Issue by prakash1998** [primefaces/primereact#1804]
- initially Toast component will have messages = [] in state
- when i call toastRef.show for the first time it's fine, and now state = { messages : [ msg1 ] }
- but after that when i call toastRef.show for 2 times back to back like this
```js
toastRef.show(msg2);
toastRef.show(msg3);
```
then it's ignoring msg2 and now state = { messages : [msg1,msg3] } ( consider msg are sticky )

Reason : it is because show method is using this.state.messages when updating state, so when i call it quickly for multiple time, then all calls have old state 

Solution: you can use fresh state which calling setState like this 
```js
this.setState(newState) // before
this.setState(prevState => newState) // after
```
i would like to submit merge request if you want me to, thanks.

**Comment by mertsincan** [comment]
Hi,

Unfortunately, I didn't fully understand this problem. Do you have a codesandbox link for us to replicate? Also, We would be grateful if you create a PR with the possible solution.

Best Regards,

**Comment by prakash1998** [comment]
here is codesandbox to regenerate the issue, https://codesandbox.io/s/awesome-johnson-csvdh
- you can excuse that i should pass array if i want to show multiple, but i'm just saying that we can resolve this issue and it's better to resolve, because it's simple example but you can think of multiple async events calling this, and some of them will be lost ( means the last most will be served to user)

**Comment by callmephil** [comment]
@prakash1998 

Ref will hold one (unique) reference of the selected HTML element. in your example, you trigger 'show' twice on the same element and that causes you to override the first call. 

Fortunately, [Primereact docs](https://primefaces.org/primereact/showcase/#/toast) explain how to triggers multiple toast/messages at the same time. it's fairly simple, you pass an array instead of a single object.

Knowing that you can refactor your function
```js
    const showMultiple = () => {
        toast.current.show(
            {severity:'info', summary:'Message 1', detail:'First message', life: 3000});
        toast.current.show(
            {severity:'info', summary:'Message 2', detail:'Second Message', life: 3000});
        
    }
```
to the following
```js
const showMultiple = () => {
       toast.current.show([
                {severity:'info', summary:'Message 1', detail:'First message'},
                {severity:'info', summary:'Message 2', detail:'Second message'},
        ]);
}
```

**Comment by prakash1998** [comment]
@callmephil i already have mentioned this solution in my previous comment, but just think about async events, which may fire at the same time, and it's not that much hard to solve as mentioned in original issue statement.

**Comment by MajorPrime** [comment]
@prakash1998 Would a "throttling" pattern work for your async use case?
All messages would be sent via a throttled function, which calls .show at most every 10ms (passing all messages received since the last call to .show)? (10ms is just an example - I don't know what value would work)

**Comment by prakash1998** [comment]
yes it works, i have tested on my side, and already using fork of it in my project temporarily.

**Comment by melloware** [comment]
I think this will be fixed by this PR: https://github.com/primefaces/primereact/pull/2669

**Comment by melloware** [comment]
This is fixed in 8.0.0-RC1 now that hooks are being used: https://codesandbox.io/s/primereact-test-forked-5zuel3
